tools/rumble/cmd/vulns.go

```diff
@@ -73,20 +71,54 @@ func (v *vulnsJson) generateJSON() error {
 	}
 	defer closer()
 
-	q := v.bqClient.Client.Query(cgbigquery.AllVulnsQuery)
-	allVulnRecords, err := v.bqClient.Query(q, cgbigquery.CveQueryType)
+	// Single query to get all vulnerability-image relationships
+	q := v.bqClient.Client.Query(cgbigquery.AllVulnsWithImagesQuery)
+	allRecords, err := v.bqClient.Query(q, cgbigquery.VulnWithImagesQueryType)
 	if err != nil {
-		log.Fatalf("error performing AllVulnsQuery: %v", err)
+		log.Fatalf("error performing AllVulnsWithImagesQuery: %v", err)
 	}
 
-	v.vulns, err = v.grypeDb.ExtractCVEs(allVulnRecords)
+	// Group results by vulnerability ID
+	vulnImageMap := make(map[string][]cgbigquery.VulnWithImages)
+	for _, record := range allRecords {
+		vwi := record.(*cgbigquery.VulnWithImages)
+		vulnImageMap[vwi.Vulnerability] = append(vulnImageMap[vwi.Vulnerability], *vwi)
+	}
+
+	// Extract unique CVE IDs for enrichment
+	var cveRecords []interface{}
```

Comment:
We know the length of this ahead of time, probably should pre-allocate so we don't need to allocate on the fly repeatedly
```suggestion
	cveRecords := make([]interface{}, 0, len(vulnImageMap))
```